pkg/cvo/metrics.go

```diff
@@ -297,6 +307,18 @@ func (m *operatorMetrics) Collect(ch chan<- prometheus.Metric) {
 		g.Set(1.0)
 		ch <- g
 	}
+
+	if availableUpdates := m.optr.getAvailableUpdates(); availableUpdates != nil {
```

Comment:
when is this `nil`?  Is that just "we were just born and haven't completed a single graph-pull cycle yet"?

Reply:
Not sure when but here's the why I check: https://github.com/openshift/cluster-version-operator/blob/2c4931dc283c551938be1a00fee290de0b79d99c/pkg/cvo/availableupdates.go#L107

Do you think we should issue a warning when it's nil?  It will spit out every 30 seconds until it gets set. Could even be an Alert if we wanted.

Reply:
I think we should figure out why it can be `nil` and update the docstring there, and *then* figure out what to do about that case for alerts.

Reply:
It appears it can only be nil if it does not get initialized. Here's the method that sets it, including initially: https://github.com/openshift/cluster-version-operator/blob/2c4931dc283c551938be1a00fee290de0b79d99c/pkg/cvo/cvo.go#L499

The first error where CVO can't find itself I would hope happens very rarely if ever and if it does perhaps the cluster is toast anyway. 

I would think the more likely scenario is ValidateClusterVersion https://github.com/openshift/cluster-version-operator/blob/2c4931dc283c551938be1a00fee290de0b79d99c/lib/validation/validation.go#L16 fails, but, don't know the circumstances or liklyhood behind that either.

Thoughts?

/cc @wking

Reply:
Both of those seem rare, and unrelated to getting updates, so probably ignore them if they happen.  Not sure if skipping the `ch <- g` means we continue to serve whatever the last value was or if it means we don't serve the metric at all.  Probably means we keep serving the last value, which is what we want for the `nil` case.

Reply:
I output a warning if availableUpdtates is null.
Set 'ch <- g' to status condition true or false. Unknown "should" never happen but if it does output a warning.